matrix/nth_fibonnacci_using_matrix_exponentiation.py

```diff
@@ -0,0 +1,98 @@
+"""
+Implementation of finding nth fibonacci number using matrix exponentiation.
+Time Complexity is about O(log(n)*8)
+As we know
+    f[n] = f[n-1] + f[n-1]
+Converting to matrix,
+    [f(n),f(n-1)] = [[1,1],[1,0]] * [f(n-1),f(n-2)]
+->  [f(n),f(n-1)] = [[1,1],[1,0]]^2 * [f(n-2),f(n-3)]
+    ...
+    ...
+->  [f(n),f(n-1)] = [[1,1],[1,0]]^(n-1) * [f(1),f(0)]
+
+So we just need the n times multiplication of  the matrix [1,1],[1,0]].
+We can decrease the n times multiplication by following  the divide and conquer approach
+
+"""
+from __future__ import print_function
+
+
+def multiply(matrix_a, matrix_b):
+    matrix_c = []
+    n = len(matrix_a)
+    for i in range(n):
+        list_1 = []
+        for j in range(n):
+            val = 0
+            for k in range(n):
+                val = val + matrix_a[i][k] * matrix_b[k][j]
+            list_1.append(val)
+        matrix_c.append(list_1)
+    return matrix_c
+
+
+def identity(n):
+    return [[int(row == column) for column in range(n)] for row in range(n)]
+
+
+def zerro(n):
+    return [[int(row == column) for column in range(n)] for row in range(n)]
+
+
+def nth_fibonacci(n):
```

Comment:
Please add some doctests of the form:
```python
    def nth_fibonacci(n):
        """
        >>> nth_fibonacci(100)
        354224848179261915075
        >>> nth_fibonacci(-100)
        -100
        """
```
And the run __python -m doctests -v matrix/nth_fibonnacci_using_matrix_exponentiation.py__ to make sure that they pass.  Also consider adding doctest to other key functions.  Our automated testing will also run these tests on each pull request.

Reply:
Added and tested locally.